src/app/provisioning-client.ts

```diff
@@ -1,28 +1,33 @@
-import { type Options } from '../types';
+import { type RpcOptions } from '@improbable-eng/grpc-web/dist/typings/client.d';
 import pb from '../gen/provisioning/v1/provisioning_pb';
 import { ProvisioningServiceClient } from '../gen/provisioning/v1/provisioning_pb_service';
-import { RobotClient } from '../robot';
-import type { Provisioning } from './provisioning';
 import { promisify } from '../utils';
-import { type CloudConfig, encodeCloudConfig } from './types';
 
-export class ProvisioningClient implements Provisioning {
-  private client: ProvisioningServiceClient;
-  private readonly options: Options;
+export type CloudConfig = pb.CloudConfig.AsObject;
 
-  constructor(client: RobotClient, options: Options = {}) {
-    this.client = client.createServiceClient(ProvisioningServiceClient);
-    this.options = options;
-  }
+const encodeCloudConfig = (obj: CloudConfig): pb.CloudConfig => {
+  const result = new pb.CloudConfig();
+  result.setId(obj.id);
+  result.setSecret(obj.secret);
+  result.setAppAddress(obj.appAddress);
+  return result;
+};
+
+export class ProvisioningClient {
+  private service: ProvisioningServiceClient;
 
-  private get service() {
-    return this.client;
+  constructor(serviceHost: string, grpcOptions: RpcOptions = {}) {
+    this.service = new ProvisioningServiceClient(serviceHost, grpcOptions);
   }
 
+  /**
+   * Get the status of the Smart Machine.
+   *
+   * @returns The Smart Machine status
+   */
   async getSmartMachineStatus() {
     const { service } = this;
     const request = new pb.GetSmartMachineStatusRequest();
-    this.options.requestLogger?.(request);
```

Comment:
Why remove the loggers?

Reply:
More in line with the other app clients (originally I originally based it on a service instead)